honeybadgermpc/protocols/avid.py

```diff
@@ -0,0 +1,216 @@
+# coding=utf-8
+import logging
+import math
+import asyncio
+from honeybadgermpc.protocols.reliablebroadcast \
+    import encode, decode, merkle_tree, get_merkle_branch, merkle_verify
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.ERROR)
+# Uncomment this when you want logs from this file.
+# logger.setLevel(logging.NOTSET)
+
+
+class AVID:
+    # Class for Asynchronous Verifiable Information Dispersal
+    def __init__(self, n, f, leader, recv, send, input_size):
+        """ Initialize the class with parameters:
+        :param int n:  at least 3
+        :param int f: fault tolerance, ``N >= 3f + 1``
+        :param int leader: ``0 <= leader < N``
+        :param recv: :func:`recv()` blocks until a message is
+            recvd; message is of the form::
+            (i, (tag, ...)) = recv()
+            where ``tag`` is one of ``{"VAL", "ECHO", "READY"}``
+        :param send: sends (without blocking) a message to a designated
+            recipient ``send(i, (tag, ...))``
+        :param input_size: the size of the input messages to be retrieved
+        """
+        assert n >= 3 * f + 1
+        assert f >= 0
+        assert 0 <= leader < n
+
+        self._data = [None] * n
+        self.n = n
+        self.f = f
```

Comment:
lets rename `f` to `t` -> the paper might say `f` but we use `t` as the threshold in most of the places in the codebase.

Reply:
Resolved.